### PR TITLE
feat(env-var): Allow to use both che-server env variables and DevWorkspace env variables

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
DevWorkspace is dropping old CHE_* env variables
So CheTheia needs to work in both mode

### What issues does this PR fix or reference?
https://github.com/devfile/devworkspace-operator/issues/344

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=next

Change-Id: Ifb177150471a4f8a818b65e6eb413fcc61a3eae2
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

**Comment by benoitf** [comment]
[crw-ci-test]

**Review by vitaliy-guliy — APPROVED** [review]
Haven't checked the behavior, but code looks good.

File: dockerfiles/theia/src/entrypoint.sh [hunk 1]
```diff
@@ -81,7 +81,8 @@ fi
 shopt -u nocasematch
 
 # run Che-Theia
-PROJ_ROOT=${CHE_PROJECTS_ROOT:-/projects}
+PROJ_ROOT=${PROJECTS_ROOT:-}
+[[ -z "$PROJ_ROOT" ]] && PROJ_ROOT=${CHE_PROJECTS_ROOT:-/projects}
```
Comment (vzhukovs) [inline dockerfiles/theia/src/entrypoint.sh:85]: As far as I understand `PROJECTS_ROOT` and `CHE_PROJECTS_ROOT` can be set in devfile?

Comment (benoitf) [reply dockerfiles/theia/src/entrypoint.sh:85]: DevWorkspace engine provides only `PROJECTS_ROOT` environment variable
Che-server, only `CHE_PROJECTS_ROOT`

if you use a devfile v1, then you can reference `CHE_PROJECTS_ROOT`, for a devfile v2, `CHE_PROJECTS_ROOT`

Comment (vzhukovs) [reply dockerfiles/theia/src/entrypoint.sh:85]: Got it! Thanks!

**Review by vzhukovs — APPROVED** [review]

File: extensions/eclipse-che-theia-plugin-ext/src/plugin/che-sidecar-content-reader.ts [hunk 2]
```diff
@@ -18,9 +18,12 @@ import { URI } from 'vscode-uri';
 export class CheSideCarContentReaderImpl implements CheSideCarContentReader {
   constructor(rpc: RPCProtocol) {
     const delegate = rpc.getProxy(PLUGIN_RPC_CONTEXT.CHE_SIDERCAR_CONTENT_READER_MAIN);
-    const machineName = process.env.CHE_MACHINE_NAME;
-    if (machineName) {
-      delegate.$registerContentReader(`file-sidecar-${machineName}`);
+    let componentName = process.env.DEVWORKSPACE_COMPONENT_NAME;
+    if (!componentName) {
+      componentName = process.env.CHE_MACHINE_NAME;
+    }
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-plugin-ext/src/plugin/che-sidecar-content-reader.ts:24]: maybe, a bit shorter variant?
`const componentName = process.env.DEVWORKSPACE_COMPONENT_NAME || process.env.CHE_MACHINE_NAME;`

File: extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote.ts [hunk 3]
```diff
@@ -38,10 +38,20 @@ process.on('unhandledRejection', (reason: any, promise: Promise<any>) => {
...
         unhandledPromises.splice(index, 1);
-        const containerName = process.env['CHE_MACHINE_NAME'];
-        console.error(`Remote plugin in ${containerName}: promise rejection is not handled in two seconds: ${err}`);
+        let componentName;
+        // If PROJECTS_ROOT is defined use it else switch to old env name
+        const devWorkspaceComponentNameVar = process.env.DEVWORKSPACE_COMPONENT_NAME;
+        if (devWorkspaceComponentNameVar) {
+          componentName = devWorkspaceComponentNameVar;
+        } else {
+          const chemachineNameEnvVar = process.env.CHE_MACHINE_NAME;
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote.ts:47]: che**M**achineNameEnvVar

**Review by azatsarynnyy — APPROVED** [review]
I've tested it with DevWorkspace. The imported projects are in place, terminal widget is working.
Just a couple of minor notes.

File: extensions/eclipse-che-theia-plugin-ext/src/plugin/che-sidecar-content-reader.ts (hunk 2 above)
Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/src/plugin/che-sidecar-content-reader.ts:24]: yes ! will update

File: extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote.ts (hunk 3 above)
Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote.ts:47]: good catch 👍

**Comment by benoitf** [comment]
@azatsarynnyy I've updated code to match your feedback

**Comment by benoitf** [comment]
7.29.0 has been tagged, merging in a couple of hours as happy path tests passed on alpine/next ans ubi8/next

**Pull request merged**